lib/include/openamp/virtio.h

```diff
@@ -111,6 +111,24 @@ struct virtio_buffer_info {
 	unsigned long size;
 };
 
+/**
+ * struct remoteproc_vring - remoteproc vring structure
+ * @vq virtio queue
+ * @va logical address
+ * @notifyid vring notify id
+ * @num_descs number of descriptors
+ * @align vring alignment
+ * @io metal I/O region of the vring memory, can be NULL
+ */
+struct virtio_vring {
+	struct virtqueue *vq;
+	void *va;
+	uint32_t notifyid;
+	unsigned int num_descs;
+	unsigned int align;
+	struct metal_io_region *io;
```

Comment:
what is this region for?

Reply:
vring area is also shared memory, and thus, it has I/O to abstract operation on the vring shared memory region.